evennia/utils/evstring.py

```diff
@@ -0,0 +1,1497 @@
+
+    def __len__(self):
+        """Overrides the default length calculation to return the visible display length"""
+        return self._visible_length
+
+    def __repr__(self):
+        if self._visible_length:
+            return f"EvCode('{str(self)}', {self._visible_length})"
+        else:
+            return f"EvCode('{str(self)}')"
+
+class EvLink(str):
+    """
+    A string-like object which represents a clickable MXP-style link. It does NOT do any splitting of the
+    input string on its own, but rather, it takes the link and the text as two separate arguments.
+    
+    The type of link can be passed via the 'link_key' kwarg, but is assumed to be an MXP command by default.
+    
+    """
+    def __new__(cls, *args, **kwargs):
+        # if it's already an EvLink, we just use its data
+        if hasattr(args[0], '_link_string'):
+            link = args[0]._link_string
+            text = args[0]._ev_string
+            key = args[0]._link_key
+            raw_string = args[0]._raw_string
+        else:
+            # the text can contain markup, so we make sure to convert it here
+            text = EvString(args[0])
+            link = kwargs.get('link_value', '')
+            key = kwargs.get('link_key', 'c')
+        
+            # ignore keys that don't match
+            if key not in ('c', 'u'):
+                key = 'c'
+
+            # recreate the original markup string, if not provided
+            if not (raw_string := kwargs.get('raw')):
+                raw_string =  f"|l{key}{link}|lt{text}|le"
+
+        # we initialize on the raw display-string text to match EvString's implementation
+        ev_link = super().__new__(EvLink, text.raw())
+        ev_link._raw_string = raw_string
+        ev_link._link_string = link
+        ev_link._link_key = key
+        ev_link._ev_string = text
+
+        return ev_link
+
+    def __repr__(self):
+        """Returns the code that would actually recreate this object"""
+        return f"EvLink('{self._ev_string.__repr__()}', link_value={self._link_string}, link_key={self._link_key}, raw='{self._raw_string}')"
+
+    def __getitem__(self, item):
+        """Returns a new EvLink with the visible text's character(s) corresponding to the index or slice"""
+        text = self._ev_string[item]
+        
+        return EvLink(text, link_value=self._link_string, link_key=self._link_key)
+    
+    def partition(self, sep, reverse=False):
+        """
+        Splits the link text on the first instance of sep. Returns a tuple of length 3 containing a link
+        with the first segment of text, the separator, and an EvString of the tail. If sep is not found,
+        returns a tuple of self followed by two empty strings.
+        """
+        if sep not in self.clean():
+            return (self, '', '')
+        
+        result = self._ev_string.partition(sep)
+        return ( EvLink(result[0], link_value=self._link_string, link_key=self._link_key), result[1], result[2] )
```

Comment:
I believe you would want to wrap `EvLink()` around all of these in order to retain the unbroken link functionality, possibly even the separator.